component/attr_cache/attr_cache.go

```diff
@@ -396,11 +345,17 @@ func (ac *AttrCache) CreateDir(options internal.CreateDirOptions) error {
 				return os.ErrExist
 			}
 			newDirAttr := internal.CreateObjAttrDir(newDirPath)
-			newDirAttrCacheItem := newAttrCacheItem(newDirAttr, true, time.Now())
+			newDirAttrCacheItem := ac.cacheMap.insert(newDirAttr, true, time.Now())
 			newDirAttrCacheItem.markInCloud(false)
-			ac.cacheMap[newDirPath] = newDirAttrCacheItem
+
 		} else {
-			ac.invalidatePath(options.Name)
+			dirAttrCacheItem, err := ac.cacheMap.get(internal.TruncateDirName(options.Name))
+			if err != nil {
+				log.Err("could not find the attr cached item: ", err)
```

Comment:
This entry will usually not exist in the cache. There is no need to log an error here, or do anything if `err` is not `nil`. So instead of an else, there should just be a single condition: `if err == nil`

Reply:
`		} else {
			dirAttrCacheItem, err := ac.cacheMap.get(internal.TruncateDirName(options.Name))
			if err == nil {
				dirAttrCacheItem.invalidate()
			}
		`

Is this what you mean to do instead?